ui/src/app/components/object-detail/object-detail.component.ts

```diff
@@ -0,0 +1,371 @@
+            maxWidth: '500px',
+          })
+        }
+      },
+    })
+  }
+
+  updateSpTableSuggestion() {
+    this.isSpTableSuggesstionDisplay = []
+    this.spTableSuggestion = []
+    this.tableData.forEach((item: any) => {
+      const srDataType = item.srcDataType
+      const spDataType = item.spDataType
+      let brief: string = ''
+      this.typeMap[srDataType].forEach((type: any) => {
+        if (spDataType == type.T) brief = type.Brief
+      })
+      this.isSpTableSuggesstionDisplay.push(brief !== '')
+      this.spTableSuggestion.push(brief)
+    })
+  }
+  spTableEditSuggestionHandler(index: number, spDataType: string) {
+    const srDataType = this.tableData[index].srcDataType
+    let brief: string = ''
+    this.typeMap[srDataType].forEach((type: any) => {
+      if (spDataType == type.T) brief = type.Brief
+    })
+    this.isSpTableSuggesstionDisplay[index] = brief !== ''
+    this.spTableSuggestion[index] = brief
+  }
+
+  toggleFkEdit() {
+    if (this.isFkEditMode) {
+      let updatedFkNames: Record<string, string> = {}
+
+      this.fkArray.value.forEach((fk: IFkTabData, i: number) => {
+        let oldFk = this.fkData[i]
+        if (oldFk.spName !== fk.spName) {
+          updatedFkNames[oldFk.spName] = fk.spName
+        }
+      })
+
+      this.data.updateFkNames(this.currentObject!.name, updatedFkNames).subscribe({
+        next: (res: string) => {
+          if (res == '') {
+            this.isFkEditMode = false
+          } else {
+            this.dialog.open(InfodialogComponent, {
+              data: { message: res, type: 'error' },
+              maxWidth: '500px',
+            })
+          }
+        },
+      })
+    } else {
+      this.currentTabIndex = 2
+      this.isFkEditMode = true
+    }
+  }
+
+  dropFk(element: any) {
+    let ind: number = this.getRemovedFkIndex(element)
+    this.data.dropFk(this.currentObject!.name, ind).subscribe({
+      next: (res: string) => {
+        if (res == '') {
+          this.data.getDdl()
+          console.log(element, 'element')
+
+          this.snackbar.openSnackBar(
+            `${element.get('spName').value} Foreign key dropped successfully`,
+            'Close',
+            5
+          )
+        } else {
+          this.dialog.open(InfodialogComponent, {
+            data: { message: res, type: 'error' },
+            maxWidth: '500px',
+          })
+        }
+      },
+    })
+  }
+
+  getRemovedFkIndex(element: any) {
+    let ind: number = -1
+
+    this.fkArray.value.forEach((fk: IFkTabData, i: number) => {
+      if (fk.spName === element.get('spName').value) {
+        ind = i
+      }
+    })
+    return ind
+  }
+  convertToFk() {
+    alert('Feature comming soon!')
+  }
+
+  checkIsInterleave() {
+    if (this.currentObject) {
+      this.data.getInterleaveConversionForATable(this.currentObject!.name)
+    }
+  }
+
+  setInterleave() {
+    this.data.setInterleave(this.currentObject!.name)
+  }
+
+  getParentFromDdl() {
```

Comment:
why do we need this to extract the parent from DDL ?
We should always default to getting it from the session/conv.

This might be error prone

Reply:
Added an item in the backlog (No - 61)